src/dex/curve-v1-factory/curve-v1-factory.ts

```diff
@@ -1187,14 +1216,70 @@ export class CurveV1Factory
       destToken,
       destAmount,
       swapData,
-      exchange,
+      this.config.router,
     );
   }
 
   static getDirectFunctionNameV6(): string[] {
     return [DIRECT_METHOD_NAME_V6];
   }
 
+  // Multihop case encoding for CurveRouter
+  // Curve Ng Router exchange function params description https://github.com/curvefi/curve-router-ng/blob/master/contracts/Router.vy#L180
+  getMultihopParam(
+    srcAmount: string,
+    destAmount: string,
+    data: CurveV1FactoryData,
+    side: SwapSide,
+  ): CurveV1RouterParam {
+    const pathLength = 11;
+    const swapParamsLength = 5;
+    const poolsLength = 5;
+
+    const path = data.path
+      .map((item, index) =>
+        index === 0
+          ? [item.tokenIn, item.exchange, item.tokenOut] // we need tokenIn only for the first item because there is no prev pool
+          : [item.exchange, item.tokenOut],
+      )
+      .flat();
+
+    while (path.length < pathLength) {
+      path.push(NULL_ADDRESS);
+    }
+
+    const swapParams: CurveV1RouterSwapParam[] = data.path.map(item => [
+      item.i,
+      item.j,
+      item.underlyingSwap
+        ? CurveRouterSwapType.exchange_underlying
+        : CurveRouterSwapType.exchange,
+      CurveRouterPoolType.stable,
+      item.n_coins,
+    ]);
+
+    while (swapParams.length < swapParamsLength) {
+      swapParams.push([0, 0, 0, CurveRouterPoolType.non_stable, 0]);
```

Comment:
What if StableNgPools are used in multhihop, will it work?

Reply:
updated

Reply:
better to pass 0 instead of CurveRouterPoolType.non_stable to avoid confusion